### PR TITLE
fix(consensus): cache unit count in `BackupWriter`

### PR DESCRIPTION
**Pull request by dpc**
Partially reverting beb5892f0fb7ac485fe47b9b5091ecd9bda2dda8 .

This query performance is not as good as we would hope.

Re #6050

### CODE REVIEW

**Comment by dpc** [comment]
Should get ack from @joschisan before landing.

**Review by joschisan — APPROVED** [review]

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.4`:
- #6057

**Pull request merged**
